src/tsk_fs_file_handler.rs

```diff
@@ -0,0 +1,170 @@
+#![allow(non_camel_case_types)]
+
+use std::io::{Read, Seek, SeekFrom};
+use std::ffi::{CStr, CString};
+use super::{
+    tsk_fs_file::TskFsFile,
+    tsk_fs_attr::TskFsAttrType,
+    errors::TskError,
+    bindings as tsk
+};
+
+#[derive(Copy, Clone)]
+pub enum TskFsFileReadFlag {
+    None,
+    Slack,
+    NoID
+}
+
+#[derive(Copy, Clone)]
+pub struct TskFsFileHandler<'fs>{
+    tsk_fs_file: &'fs TskFsFile<'fs>,
+    _offset: i64,
+    attr_size: i64,
+    id: u16,
+    attr_to_read: TskFsAttrType,
+    pub read_flag: TskFsFileReadFlag
+}
+
+/// TskFsFileHandler struct is used to read file data.
+impl<'fs> TskFsFileHandler<'fs> {
+    /// Create TskFsFileHandler from TskFsFile. This will default to read $DATA attribute.
+    pub fn new(tsk_fs_file: &'fs TskFsFile<'fs>) -> Result<Self,TskError> {
```

Comment:
I think `new` will need to have a `TskFsFile`, `TskFsAttr`, and read flag passed in.